crates/requirements-txt/src/lib.rs

```diff
@@ -317,6 +317,27 @@ pub struct RequirementsTxt {
     pub no_index: bool,
 }
 
+/// Calculates column and line based on the cursor and content.
+fn calculate_line_column_pair(content: &str, position: usize) -> (usize, usize) {
+    let mut line = 1;
+    let mut column = 1;
+
+    for (index, char) in content.char_indices() {
+        if index >= position {
+            break;
+        }
+        // This should work fine for both Windows and Linux line endings
+        if char == '\n' {
```

Comment:
So here, if we see `\r\n`, I think we technically want to avoid incrementing `column` after we see the `\r`. I can't see how this would matter in practice for _this_ use-case, but e.g., you can see an example of how that's done here in Ruff: https://github.com/astral-sh/ruff/blob/c9931a548ff07c031130571f3664343bea224026/crates/ruff_source_file/src/line_index.rs#L29

Reply:
>  I think we technically want to avoid incrementing column after we see the \r

Agreed, I left this as-is on purpose but I really debated to either keep this for simplicity or track the prev_char reference for this check. Luckily it can be changed easily whichever route we want to go.

Reply:
I'd prefer to change it just for completeness, in case this gets reused elsewhere. Are you ok to modify it?

Reply:
Let me know if this is what you had in mind d0f7161f1c4b2e903f73256b51368d32dbb34cc5

Reply:
I was aiming for something more like: if we see `\r`, check if the next character is `\n`; if so, skip it. (That would correctly handle `\r`, `\n`, and `\r\n`.

Reply:
Sorry for the delay (work 😆), hopefully this is closer 223fd99ac1b44e0557227d1c461a20bf1a9d85f2